p2p/tests/kademlia-incoming.rs

```diff
@@ -0,0 +1,78 @@
+use p2p_testing::{
+    cluster::{ClusterBuilder, Listener},
+    rust_node::RustNodeConfig,
+};
+use std::time::Duration;
+
+#[tokio::test]
+#[ignore]
+async fn kademlia_incoming() {
+    let mut cluster = ClusterBuilder::new()
+        .ports(11000..11200)
+        .idle_duration(Duration::from_millis(100))
+        .start()
+        .await
+        .expect("should build cluster");
+
+    let node1 = cluster
+        .add_rust_node(RustNodeConfig::default())
+        .expect("node1");
+
+    let config = RustNodeConfig {
+        initial_peers: vec![Listener::Rust(node1)],
+        ..Default::default()
+    };
+
+    let node2 = cluster.add_rust_node(config).expect("node2");
+
+    let peer_id1 = cluster.rust_node(node1).state().my_id();
+    let peer_id2 = cluster.rust_node(node2).state().my_id();
+
+    let bucket = &cluster
+        .rust_node(node2)
+        .state()
+        .network
+        .scheduler
+        .discovery_state()
+        .expect("Must be ready")
+        .routing_table
+        .buckets
+        .first()
+        .expect("There must be at least 1 bucket");
+
+    assert_eq!(
+        bucket.len(),
+        2,
+        "There must be 2 items in bucket self and seed node"
+    );
```

Comment:
Hmm... does it really work for you?
The thing with this testing cluster is that unless you poll its stream, nothing will happen, no connection, no kad messages exchange.

Reply:
Test does work, and first node is in seconds node routing table, but I am not sure how, since `effects` from `P2pNetworkKadAction` never get called.

Reply:
Ah, that explains things. Indeed, Kademlia table is populated with initial peers on creation. Well, that makes sense to test as well, but more complex case where routing is updated during Kademlia bootstrap is more important,